src/aura/autocomplete/autocomplete.cmp

```diff
@@ -77,6 +77,7 @@
                                   class="{!'slds-input ' + (and(v.showPills, !empty(v.value), !empty(v.displayValue)) ? 'slds-hide' : '')}"
                                   placeholder="{!v.placeholder}"
                                   value="{!v.displayValue}"
+                                  updateOn="keyup"
```

Comment:
you were hooking up the 'input' event which is not an official Lightning event.  thus one could not look at component.get('v.value') in the event handler; it wasn't set.  So first I changed it to be the 'change' event, but turns out Lightning only calls that when the textbox loses focus, rather than on each key.  So I had to go with keyup instead.